frontend/src/metabase/dashboard/components/Dashboard/Dashboard.tsx

```diff
@@ -513,51 +534,58 @@ function DashboardInner(props: DashboardProps) {
       loading={!dashboard}
       error={error}
     >
-      {() => (
-        <DashboardStyled>
-          <DashboardHeaderContainer
-            data-element-id="dashboard-header-container"
-            id="Dashboard-Header-Container"
-            isFullscreen={isFullscreen}
-            isNightMode={shouldRenderAsNightMode}
-          >
-            {/**
-             * Do not conditionally render `<DashboardHeader />` as it calls
-             * `useDashboardTabs` under the hood. This hook sets `selectedTabId`
-             * in Redux state which kicks off a fetch for the dashboard cards.
-             */}
-            <DashboardHeader
-              {...props}
-              onEditingChange={handleSetEditing}
-              setDashboardAttribute={handleSetDashboardAttribute}
-              addParameter={addParameter}
-              onSharingClick={handleToggleSharing}
-            />
-          </DashboardHeaderContainer>
-
-          <DashboardBody isEditingOrSharing={isEditing || isSharing}>
-            <ParametersAndCardsContainer
-              id={DASHBOARD_PDF_EXPORT_ROOT_ID}
-              data-element-id="dashboard-parameters-and-cards"
-              data-testid="dashboard-parameters-and-cards"
-              shouldMakeDashboardHeaderStickyAfterScrolling={
-                !isFullscreen && (isEditing || isSharing)
-              }
+      {() => {
+        if (!dashboard) {
+          return null;
+        }
+
+        return (
+          <DashboardStyled>
+            <DashboardHeaderContainer
+              data-element-id="dashboard-header-container"
+              id="Dashboard-Header-Container"
+              isFullscreen={isFullscreen}
+              isNightMode={shouldRenderAsNightMode}
             >
-              {renderParameterList()}
-              <CardsContainer data-element-id="dashboard-cards-container">
-                {renderContent()}
-              </CardsContainer>
-            </ParametersAndCardsContainer>
-
-            <DashboardSidebars
-              {...props}
-              setDashboardAttribute={handleSetDashboardAttribute}
-              onCancel={() => setSharing(false)}
-            />
-          </DashboardBody>
-        </DashboardStyled>
-      )}
+              {/**
+               * Do not conditionally render `<DashboardHeader />` as it calls
+               * `useDashboardTabs` under the hood. This hook sets `selectedTabId`
+               * in Redux state which kicks off a fetch for the dashboard cards.
+               */}
```

Comment:
(I think we should be fine here as only `if (!dashboard)` is the only condition, and that condition shouldn't change too much)